kernel/mellanox_ethernet/src/command_queue.rs

```diff
@@ -0,0 +1,740 @@
+            0x101   => {CommandOpcode::QueryAdapter}, 
+            0x102   => {CommandOpcode::InitHca}, 
+            0x103   => {CommandOpcode::TeardownHca}, 
+            0x104   => {CommandOpcode::EnableHca}, 
+            0x105   => {CommandOpcode::DisableHca}, 
+            0x107   => {CommandOpcode::QueryPages}, 
+            0x108   => {CommandOpcode::ManagePages}, 
+            0x10A   => {CommandOpcode::QueryIssi}, 
+            0x10B   => {CommandOpcode::SetIssi}, 
+            _       => {CommandOpcode::Unknown}
+        }
+    }
+
+    /// Returns the first 16 bytes of output data that are written inline in the command.
+    /// The valid values for each field are different for every command, and can be taken from Chapter 23 of the PRM.
+    fn get_output_inline_data(&self) -> (u8, u32, u32, u32) {
+        (   
+            (self.command_output_status.read().get() >> 24) as u8,
+            self.command_output_syndrome.read().get(),
+            self.command_output_inline_data_0.read().get(),
+            self.command_output_inline_data_1.read().get()
+        )
+    }
+   
+    /// Sets length of output data in bytes.
+    /// This value is different for every command, and can be taken from Chapter 23 of the PRM. 
+    fn set_output_length_in_bytes(&mut self, length: u32) {
+        self.output_length.write(U32::new(length));
+    }
+
+    /// Sets the token of the command.
+    /// This is a random value that should be the same in the command entry and all linked mailboxes.
+    fn set_token(&mut self, token: u8) {
+        let val = self.token_signature_status_own.read().get();
+        self.token_signature_status_own.write(U32::new(val | ((token as u32) << 24)));
+    }
+
+    /// Returns the status of command delivery.
+    /// This only informs us if the command was delivered to the NIC successfully, not if it was completed successfully.
+    pub fn get_delivery_status(&self) -> CommandDeliveryStatus {
+        let status = self.token_signature_status_own.read().get() & 0xFE;
+        if status == 0 {
+            CommandDeliveryStatus::Success
+        } else if status == 1 {
+            CommandDeliveryStatus::SignatureErr
+        } else if status == 2 {
+            CommandDeliveryStatus::TokenErr
+        } else if status == 3 {
+            CommandDeliveryStatus::BadBlockNumber
+        } else if status == 4 {
+            CommandDeliveryStatus::BadOutputPointer
+        } else if status == 5 {
+            CommandDeliveryStatus::BadInputPointer
+        } else if status == 6 {
+            CommandDeliveryStatus::InternalErr
+        } else if status == 7 {
+            CommandDeliveryStatus::InputLenErr
+        } else if status == 8 {
+            CommandDeliveryStatus::OutputLenErr
+        } else if status == 9 {
+            CommandDeliveryStatus::ReservedNotZero
+        } else if status == 10 {
+            CommandDeliveryStatus::BadCommandType
+        } else {
+            CommandDeliveryStatus::Unknown
```

Comment:
similar to my above comment, use a conversion trait (e.g., `From`) that incorporates a match statement instead of a series of if...else statements.